src/components/HistoricalForecast/HistoricalForecast.tsx

```diff
@@ -1,6 +1,95 @@
-import DateRangePicker from 'components/DateRangePicker';
-import * as classes from './styles';
+import { useContext } from 'react';
+// import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+import { Field, Formik, Form } from 'formik';
+import * as yup from 'yup';
+import SelectField from 'components/SelectField';
+import LANGUAGE_OPTIONS from 'constants/languageOptions';
+import Languages from 'constants/languages';
+import TEMPERATURE_UNITS_OPTIONS from 'constants/temperatureUnitsOptions';
+import TemperatureUnits from 'constants/temperatureUnits';
+import SettingsContext from 'contexts/SettingsContext';
+import Location from 'types/location';
+import SettingsContextType from 'types/settingsContextType';
+
+const LANGUAGE_CHOICE_LABEL_ID = 'language-label';
+const LANGUAGE_CHOICE_LABEL = 'Language';
+// const LOCATION_AUTOCOMPLETE = 'location-select';
+const TEMPERATURE_UNITS_LABEL_ID = 'unit-label';
+const TEMPERATURE_UNITS_LABEL = 'Temperature unit';
+
+interface Validation {
+  language: Languages;
+  startDate: Date;
+  endDate: Date;
+  location: Location;
+  temperatureUnit: TemperatureUnits;
+}
+
+const VALIDATION_SCHEMA: yup.ObjectSchema<Validation> = yup.object().shape({
+  language: yup.mixed<Languages>().required('Required'),
+  startDate: yup.date().required('Required'),
+  endDate: yup.date().required('Required'),
+  location: yup.mixed<Location>().required('Required'),
+  temperatureUnit: yup.mixed<TemperatureUnits>().required('Required'),
+});
+
+interface LanguageOption {
+  label: string;
+  value: Languages;
+}
+
+interface TemperatureUnitOption {
+  label: string;
+  value: TemperatureUnits;
+}
 
 export default function HistoricalForecast() {
-  return <DateRangePicker css={classes.dateRangePicker} />;
+  const { selectedLanguage, selectedTemperatureUnit } =
+    useContext<SettingsContextType>(SettingsContext);
+  const INITIAL_FORM_VALUES = {
+    language: selectedLanguage,
+    startDate: Date,
+    endDate: Date,
+    location: {},
+    temperatureUnit: selectedTemperatureUnit,
+  };
+
+  return (
+    <Formik
```

Comment:
Add generic for the `Formik`. Use `InferType` from `yup` in order to derive type from validation schema